### ISSUE TITLE
Roles not detected for project with collections layout

### ISSUE THREAD

**Issue by nre-ableton** [ansible-community/ansible-lint#1409]
##### Summary

Following the advice given in https://github.com/ansible-community/ansible-lint/issues/1367#issuecomment-780636162, I restructured my project to conform to the collections layout. Now, ansible-lint throws `internal-error the role 'ableton.internal-role-name' was not found in <paths list>`.

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details

```console (paste below)
$ ansible --version
ansible 2.10.5
  config file = /home/nre/<redacted>/ansible.cfg
  configured module search path = ['/home/nre/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/nre/<redacted>/lib/python3.8/site-packages/ansible
  executable location = /home/nre/Code/caches/venv/bin/ansible
  python version = 3.8.5 (default, Jul 28 2020, 12:59:40) [GCC 9.3.0]

$ ansible-lint --version
ansible-lint 5.0.2 using ansible 2.10.5
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### OS / ENVIRONMENT

Linux Pop!_OS 20.04

##### STEPS TO REPRODUCE

My project layout is roughly like this:

```
.
├── ansible
│   ├── playbooks
│   │   ├── files
│   │   ├── group_vars
│   │   ├── host_vars
│   │   ├── inventory
│   │   ├── tasks
│   │   ├── templates
│   │   ├── vars
│   │   └── vault
│   ├── roles
│   │   ├── ableton.internal-role-name
│   │   ├── geerlingguy.docker
│   │   └── geerlingguy.homebrew
│   └── vault
├── scripts
└── tests
```

We are running ansible-lint with `ansible-lint -c .ansible-lint.yml -vvv ansible` from the project's top-level directory, which results in the error:

```
ansible/playbooks/example-playbook-name.yml:42:7: internal-error the role 'ableton.internal-role-name'
was not found in /home/nre/<project name>/ansible/playbooks/roles:/home/nre/.ansible/roles:
/usr/share/ansible/roles:/etc/ansible/roles:/home/nre/<project name>/.cache/roles:
/home/nre/<project name>/ansible/playbooks
```

(Linebreaks added for readability)

It seems that `ansible-lint` is looking in `ansible/playbooks/roles` (and a bunch of other default locations), but not `ansible/roles`. 

##### Desired Behaviour

I believe that `ansible-lint` should be looking in `ansible/roles` because:

- The [collections structure](https://docs.ansible.com/ansible/latest/dev_guide/developing_collections.html#collection-structure) says that `playbooks` and `roles` should be sibling directories
- The project's `ansible.cfg` contains the line `roles_path = ./ansible/roles` in the `[defaults]` section, though I am guessing `ansible-lint` isn't parsing that file, though it would be nice if it could

I am working around this with the (undocumented?) env var `ANSIBLE_ROLES_PATH`, but I don't feel like this is a great solution. I would be happy to submit a PR, if you would agree that this behavior is desirable.

**Comment by ssbarnea** [comment]
[collections structure](https://docs.ansible.com/ansible/latest/dev_guide/developing_collections.html#collection-structure) never advertised that this layout could be in any other location than the root of the code repository. 

Digging inside code repos to find things that would look like a collection or not would improve serious performance issues so for the moment the only layout recognized is when the collection is at the root of the source repository.

Your workarounds are valid. For the moment I do not see this to be changing but if someone makes a pull request to add an option to override the default use of "cwd" as the reference location, I would not oppose it. I can see an option named `project_dir` that you be default empty and in your case could be set to `ansible`. The tool will chdir to it when running, so it will be able to find the collection, map the roles folder and so on.

**Comment by nre-ableton** [comment]
@ssbarnea Ok, that makes sense, thanks for the explanation!

I like your idea of the `project_dir` option, and maybe I'll implement that when I have a spare moment in the future.

Also, I should note that we tried running `ansible-lint` inside of the `ansible` folder, but we ran into problems with this because the `ansible.cfg` file lives one level up, so `ansible-playbook --syntax-check` fails. Therefore I would agree that setting `ANSIBLE_ROLES_PATH` is the best approach for us now.
